pkg/vpc/vpc.go

```diff
@@ -422,10 +445,10 @@ func ImportAllSubnets(provider api.ClusterProvider, spec *api.ClusterConfig) err
 			return err
 		}
 	}
-	if err := ImportSubnetsFromList(provider.EC2(), spec, api.SubnetTopologyPrivate, spec.PrivateSubnetsWithIDs(), spec.PrivateSubnetsWithCIDRs()); err != nil {
+	if err := ImportSubnetsFromList(provider.EC2(), spec, api.SubnetTopologyPrivate, spec.PrivateSubnetsWithIDs(), spec.PrivateSubnetsWithCIDRs(), spec.PrivateSubnetsWithAZs()); err != nil {
 		return err
 	}
-	if err := ImportSubnetsFromList(provider.EC2(), spec, api.SubnetTopologyPublic, spec.PublicSubnetsWithIDs(), spec.PublicSubnetsWithCIDRs()); err != nil {
+	if err := ImportSubnetsFromList(provider.EC2(), spec, api.SubnetTopologyPublic, spec.PublicSubnetsWithIDs(), spec.PublicSubnetsWithCIDRs(), spec.PrivateSubnetsWithAZs()); err != nil {
```

Comment:
I think you meant:

```suggestion
	if err := ImportSubnetsFromList(provider.EC2(), spec, api.SubnetTopologyPublic, spec.PublicSubnetsWithIDs(), spec.PublicSubnetsWithCIDRs(), spec.PublicSubnetsWithAZs()); err != nil {
```

Reply:
thanks, yes and exactly this is why I have a refactor PR ready to go

Reply:
thanks, yes and exactly this is why I have a refactor PR ready to go